cishouseholds/merge.py

```diff
@@ -239,27 +243,27 @@ def assign_time_difference_and_flag_if_outside_interval(
 
     return df.withColumn(
         column_name_outside_interval_flag,
-        F.when(~F.col(column_name_time_difference).between(interval_lower_bound, interval_upper_bound), 1).otherwise(
-            None
-        ),
+        F.when(~F.col(column_name_time_difference).between(interval_lower_bound, interval_upper_bound), 1)
+        .otherwise(None)
+        .cast("integer"),
     )
 
 
-def assign_unique_identifier_column(df: DataFrame, column_name_to_assign: str, ordering_columns: list):
-    """
-    Derive column with unique identifier for each record.
-    Parameters
-    ----------
-    df
-    column_name_to_assign
-        Name of column to be created
-    ordering_columns
-        Columns to define order of records to assign an integer value from 1 onwards
-        This order is mostly for comparison/proving purposes with stata output
-    """
-
-    window = Window.orderBy(*ordering_columns)
-    return df.withColumn(column_name_to_assign, F.row_number().over(window))
+# def assign_unique_identifier_column(df: DataFrame, column_name_to_assign: str, ordering_columns: list):
```

Comment:
We can always find old code through git, so it's safe to delete rather than commenting out